include/picongpu/fields/incidentField/profiles/ExpRampWithPrepulse.hpp

```diff
@@ -30,276 +30,290 @@
+                 * @return incident field E value in internal units
                  */
-                template<typename T_Params>
-                struct GetFunctorIncidentE<profiles::ExpRampWithPrepulse<T_Params>>
+                HDINLINE float3_X operator()(floatD_X const& totalCellIdx) const
                 {
-                    using type = profiles::detail::ExpRampWithPrepulseFunctorIncidentE<T_Params>;
-                };
+                    return Base::operator()(*this, totalCellIdx);
+                }
 
-                /** Get type of incident field B functor for the exponential ramp with prepulse  profile type
+                /** Get time-dependent longitudinal scalar factor for the given time
                  *
-                 * Rely on SVEA to calculate value of B from E.
+                 * Interface required by Base.
+                 * Gaussian transversal profile not implemented in this class, but provided by Base.
                  *
-                 * @tparam T_Params parameters
+                 * @param time time moment to calculate the factor at
+                 * @param phaseShift additional phase shift to add on top of everything else,
+                 *                   in radian
                  */
-                template<typename T_Params>
-                struct GetFunctorIncidentB<profiles::ExpRampWithPrepulse<T_Params>>
+                HDINLINE float_X getLongitudinal(float_X const time, float_X const phaseShift) const
+                {
+                    auto const runTimeShifted = time + Unitless::time_start_init;
+                    auto const phase = Unitless::w * runTimeShifted + Unitless::LASER_PHASE + phaseShift;
+                    return math::sin(phase) * Unitless::AMPLITUDE * Envelope::getEnvelope(runTimeShifted);
+                }
+            };
+        } // namespace detail
+
+        template<typename T_Params>
+        struct ExpRampWithPrepulseLongitudinal : public detail::ExpRampWithPrepulsesLongitudinalUnitless<T_Params>
+        {
+        public:
+            using Unitless = detail::ExpRampWithPrepulsesLongitudinalUnitless<T_Params>;
+
+            static constexpr float_X TIME_SHIFT = Unitless::time_start_init;
+
+            HDINLINE static float_X getEnvelope(float_X const runTime)
+            {
+                constexpr auto int_ratio_prepule = Unitless::INT_RATIO_PREPULSE;
+                constexpr auto int_ratio_point_1 = Unitless::INT_RATIO_POINT_1;
+                constexpr auto int_ratio_point_2 = Unitless::INT_RATIO_POINT_2;
+                constexpr auto int_ratio_point_3 = Unitless::INT_RATIO_POINT_3;
+                auto const AMP_PREPULSE = math::sqrt(int_ratio_prepule);
+                auto const AMP_1 = math::sqrt(int_ratio_point_1);
+                auto const AMP_2 = math::sqrt(int_ratio_point_2);
+                auto const AMP_3 = math::sqrt(int_ratio_point_3);

```

Comment:
should `int_ratio_point_...` be checked for >= 0? 
I do not think this is necessary because then the `AMP_...` just becomes `nan`.

Reply:
Code that existed before